[Murali Shruthi] iP

Shruthiiii03 opened the pull request:
# Prince Updates

> "Your mind is for having ideas, not holding them." – David Allen ([source](https://www.google.com/url?sa=t&source=web&rct=j&opi=89978449&url=https://www.linkedin.com/pulse/your-mind-having-ideas-holding-them-david-allen-paul-miklautsch&ved=2ahUKEwj2-I_DtaSIAxWuzDgGHdfFII8QFnoECBYQAQ&usg=AOvVaw3bp3UVJYsMUsqhp9xt7a7w))

Prince is **YOUR SECOND BRAIN**; it frees your mind of having to remember things you need to do. It's,

- Text-based
- Easy learning curve
- SUPER FAST performance 🚀

## YOU can do so by doing these steps : 

1. Download it from [here] (https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
2. Double-click it.
3. Add your tasks and whatever deadlines you need to keep track of.
4. Leave the rest to Prince to manage your tasks for you!

And it is **FREE**!

## Features include

- [x] Managing tasks
- [x] Managing deadlines
- [x] Managing various task types
- [ ] Reminders (coming soon)

If you're a Java programmer, you can use it to practice Java as well. Here's an example of the main method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Shruthiiii03 committed:
Level 0

Shruthiiii03 committed:
Level 1 task completed

Shruthiiii03 committed:
Level 2 task completed

Shruthiiii03 committed:
Level 3 task completed

Shruthiiii03 committed:
Completed Level 4 tasks

Shruthiiii03 committed:
Automated testing done

Shruthiiii03 committed:
Completed Level 5 tasks

Shruthiiii03 committed:
Completed Level 6 tasks

Shruthiiii03 committed:
Completed Enums

Shruthiiii03 committed:
Level 7 tasks are completed

Shruthiiii03 committed:
Completed Level-8 tasks

Shruthiiii03 committed:
Merge Level-8 branch

Shruthiiii03 committed:
Completed OOP

Shruthiiii03 committed:
Merge OOP branch

Shruthiiii03 committed:
Completed A-Packages

Shruthiiii03 committed:
Merge Package branch

Shruthiiii03 committed:
testing file

Shruthiiii03 committed:
Merge branch 'add-gradle-support'

Shruthiiii03 committed:
Completed gradle

Shruthiiii03 committed:
Merge gradle

Shruthiiii03 committed:
Completed JUnit

Shruthiiii03 committed:
Merge JUnit

Shruthiiii03 committed:
Completed JavaDoc

Shruthiiii03 committed:
Merge Javadoc branch

Shruthiiii03 committed:
Completed codingstandards

Shruthiiii03 committed:
Merge Coding standard branch

Shruthiiii03 committed:
Completed Level-9 tasks

Shruthiiii03 committed:
Merge Level-9 branch

BuffWuff1712 commented on src/main/java/exception/IncompleteDescException.java:
I like how you separated each possible exception into its own distinct exception class. This approach enhances code clarity and maintainability👍

BuffWuff1712 commented on src/main/java/ui/Ui.java:
for the function diffListDisplay, perhaps you might want to use a more intuitive function name that can explain how this list function differs from the 'list all tasks' listDisplay?

BuffWuff1712 reviewed (commented):
Overall, the code is clean and easy to understand. Great work👍

colinac commented on src/main/java/ui/Ui.java:
```diff
@@ -0,0 +1,85 @@
+    /**
+     * Generates a string displaying all tasks in the list.
+     * @param list
+     * @return String format of tasks
```
Perhaps add a period after the parameter description?

colinac commented on src/main/java/task/ToDoTask.java:
```diff
@@ -0,0 +1,44 @@
+
+    /**
+     * Constructs a ToDoTask with the specified description
+     * @param description
```
Add an empty line between description and parameter section?

colinac commented on src/main/java/ui/Ui.java:
```diff
@@ -0,0 +1,85 @@
+    /**
+     * Prints a termination message to the console.
+     */
+
```
Remove blank line between the documentation block and the method?

colinac reviewed (commented):
The code adheres to the coding standards except for that for JavaDoc comments as outlined [here](https://se-education.org/guides/conventions/java/intermediate.html#comments). Great job!👍

Shruthiiii03 committed:
Complete Level-10 tasks

Shruthiiii03 committed:
Merge Level-10 branch

Shruthiiii03 committed:
Current task validation lacks proper assertionsand it is important to verify the correctness of inputs and parameters, as well as limits to certain variables. This could lead to invalid or inappropiate task states or unhandled edge cases which could lead to issues during runtime when a user is using the application. Hence, assertions are implemented to validate the correctness and maintain consistency, and helps to catch errors early so there will be no issues during runtime.

Shruthiiii03 committed:
Merge pull request #1 from Shruthiiii03/branch-A-Assertions

Current task validation lacks proper assertionsand it is important to…

Shruthiiii03 committed:
Improve code quality by refactoring repeated code

Refactored code across different classes and methods to make code more readable and consolidated code into helper functions to clean up code.

Made sure classes and methods were appropriately named and inheritance was followed whenever possible to reduce duplicated logic. Also removed unecessary nesting whenever possible.

These changes will help make the code more readable and easier to understand for other developersin the future.

Shruthiiii03 committed:
Merge pull request #3 from Shruthiiii03/branch-A-CodeQuality

Improve code quality by refactoring repeated code

Shruthiiii03 committed:
Added the "archive" and "archivelist" commands for task management.

Users can use the "archive" command to clear tasks from their list, while saving tasks to the archive for future reference. Using the "archivelist" command, users can view their tasks in the archived list.

This helps users to start their todo list on a clean slate while retaining past tasks in the archive.

Shruthiiii03 committed:
Merge pull request #4 from Shruthiiii03/branch-C-Archive

Added the "archive" and "archivelist" commands for task management.

Shruthiiii03 committed:
Added more aesthetic profile pictures

Shruthiiii03 committed:
Merge pull request #5 from Shruthiiii03/branch-A-BetterGui

Added more aesthetic profile pictures

Shruthiiii03 committed:
Included more error handling

Shruthiiii03 committed:
Merge pull request #6 from Shruthiiii03/branch-A-MoreErrorHandling

Included more error handling

Shruthiiii03 committed:
Added more tests for more comprehensive testing procedure

Shruthiiii03 committed:
Merge pull request #7 from Shruthiiii03/branch-A-MoreTesting

Added more tests for more comprehensive testing procedure

Shruthiiii03 committed:
Added more J-Unit tests to test out methods and exceptions and ensure robustness of the code.

*Added Javadoc comments to improve code documentation and give user a summary of the method/class
*Added and enhanced error messages to guide users through the application and for debugging

These changes aim to increase test coverage and cover potential issues that could rise during runtime, making the codebase more reliable and easier to understand.

Shruthiiii03 committed:
EventTask's start and end times are now represented as LocalDateTime objects instead of Strings.

Previously, the start and end times were handled as strings, which required manual saving of the duration of the event task. After using LocalDateTime, the class can automatically parse the input duration into dates and times and this improves the efficiency of the bot.

*Updated EventTask constructor to parse date-time strings into LocalDateTime.
*Added dateTimeFormatter for consistent formatting of time and date throughout the application.
*Modified printTask and printFileFormat to suit the new data types of the start and end time of Event Tasks.

These changes improve the robustness of date-time handling and improves the efficiency of task management.

Shruthiiii03 committed:
TaskListTest and ParserTest have been enhanced with more testing and JavaDocs.

-Added JavaDoc comments to the TaskListTest and ParserTest class to improve readability and documentation.
-Enhanced ParserTest class to include more comprehensive testing.
-Updated test cases for adding and deleting tasks in TaskList and included negative test cases for handling unknown commands in ParserTest.

Shruthiiii03 committed:
Added error handling for empty task descriptions

-Added a check for null or empty command in inputs (including multiple spaces)
-Throws IncompleteDescException with an error message to alert users to input a proper command

Shruthiiii03 committed:
Adhered to coding standards and improved code quality

-Reviewed and refined code to follow Java and Git conventions.
-Removed unnecessary commented-out code for improved clarity.
-Added spaces wherever necessary to improve code readability.
-Ensured the absence of very-long methods or deeply nested code.

Enhanced code quality and code readability and finalised all the important features of Prince.

Shruthiiii03 committed:
Merge pull request #8 from Shruthiiii03/branch-A-FinaliseFeatures

Branch-A-FinaliseFeatures is completed and IP is completed.

Shruthiiii03 committed:
Enhanced user guide in README and added product screenshot (Ui.png) to docs

-Made README more comprehensive to make it easier for users to navigate through Prince

Shruthiiii03 committed:
Merge pull request #9 from Shruthiiii03/branch-A-UserGuide

Enhanced user guide in README and added product screenshot (Ui.png) t…

Shruthiiii03 committed:
Made all final edits

-Included citations
-Checked all code and its functionalities
-Ready for submission

Shruthiiii03 committed:
Added Ui.png

Added the product demo screenshot for users to view Prince bot.

Shruthiiii03 committed:
Update README.md

Updated README to add in images that demo commands to users
- Finalised commands 
- Added pictures

Shruthiiii03 committed:
Have finalised all code

-Updated README
-Updated saving of tasks in the file in other computers
-Changed formatting of GUI in an attempt to remove errors
-Finalised IP

Shruthiiii03 committed:
Edited the dependencies to resolve a few errors

-Edited and kept only the JAVAFX dependencies for mac

Shruthiiii03 committed:
Added some error-checking for deadline and event tasks

-Added an invalid deadline exception for deadline and event commands with no deadline

Shruthiiii03 committed:
Had to update ReadME changes before committing changes to github

Shruthiiii03 committed:
Add JavaDoc comments to ArchiveListCommand.java

Shruthiiii03 committed:
Remove unnecessary imports in Command.java

Shruthiiii03 committed:
Remove unnecessary imports and comments in GeneralTaskCommand.java

Shruthiiii03 committed:
Add JavaDoc comments in ToDoCommand.java

Shruthiiii03 committed:
Improve code style and quality

Several classes have unnecessary imports, missing spacing between imports of
the same package, and unnecessary comments.

Let's:
* remove unnecessary imports and comments
* add a line of spacing between imports of the same package

Shruthiiii03 committed:
Add spacing for imports in MainWindow.java

Shruthiiii03 committed:
Clean up JavaDoc comments in Exception package

Shruthiiii03 committed:
Remove unnecesary comments from Storage.java
